proms-resources/backend/src/main/java/io/uhndata/cards/proms/internal/importer/ImportEndpoint.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.uhndata.cards.proms.internal.importer;
+
+import java.io.IOException;
+import java.io.Writer;
+
+import javax.servlet.Servlet;
+
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.ResourceResolverFactory;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.apache.sling.servlets.annotations.SlingServletResourceTypes;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+@Component(service = { Servlet.class })
+@SlingServletResourceTypes(
+    resourceTypes = { "cards/SubjectsHomepage" },
+    selectors = { "import" })
+public class ImportEndpoint extends SlingSafeMethodsServlet
+{
+    @Reference
+    private ResourceResolverFactory resolverFactory;
+
+    @Override
+    public void doGet(final SlingHttpServletRequest request, final SlingHttpServletResponse response) throws IOException
+    {
+        final Writer out = response.getWriter();
+
+        //Ensure that this can only be run when logged in as admin
+        final String remoteUser = request.getRemoteUser();
+        if (remoteUser == null || !"admin".equals(remoteUser)) {
```

Comment:
Checking if `null` seems unnecessary here.
```suggestion
        if (!"admin".equals(remoteUser)) {
```

Reply:
Per [HttpServletRequests](https://docs.oracle.com/javaee/6/api/javax/servlet/http/HttpServletRequest.html#getRemoteUser()), `remoteUser` can be `null` if the user is not logged in.

Reply:
But that's covered by `!"admin".equals(remoteUser)`.

Reply:
Added in `c58d36f3` :heavy_check_mark:

Reply:
Doesn't seem to have been pushed yet :)